### ISSUE TITLE
CI able to pull in sonatype snapshots (even for removed modules)

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#8897]
We've had a couple of situations recently where a module was removed, but the dependency was still used/referenced somewhere.
Also a couple of modules were relocated (moved within another module) but the pom.xml modules section was never updated (hence the sub-modules were never built - effectively removed)

The problem is: the build was fine both locally (due to local maven install and/or snapshots), and on CI (due to pulling in an old copy from snapshots).

Basically, for CI: We need snapshots enabled for some libraries (notably javacpp) but not others (anything built in the deeplearning4j mono repo).
We want to avoid not being able to detect a removed/renamed/relocated module until release time - which is what is happening now.
